net/Cargo.toml

```diff
@@ -24,6 +25,7 @@ ordermap = { workspace = true, features = ["std"] }
 serde = { workspace = true, features = ["derive", "std"] }
 thiserror = { workspace = true }
 tracing = { workspace = true }
+id = { workspace = true }
```

Comment:
Should this dependency be marked as optional? It's only pulled if we build with feature `netdevsim`, isn't it?

Reply:
oh, that does make some sense.  LTO will remove all traces of the dependency if it isn't linked in at all, so technically we already have that met for release builds.  But it is likely a good idea anyway just to document why we need the ID package